src/parser/shared/modules/features/LowHealthHealing/Component.tsx

```diff
@@ -1,30 +1,128 @@
 import { Trans } from '@lingui/macro';
 import { formatNumber, formatPercentage, formatDuration } from 'common/format';
-import SPECS from 'game/SPECS';
 import { SpellLink } from 'interface';
-import { Icon } from 'interface';
 import { SpecIcon } from 'interface';
 import { TooltipElement } from 'interface';
-import PropTypes from 'prop-types';
+import Combatant from 'parser/core/Combatant';
+import { Ability, HealEvent } from 'parser/core/Events';
+/* eslint-disable */
+// @ts-ignore
 import Slider from 'rc-slider';
-import { Component } from 'react';
-
+/* eslint-enable */
+import * as React from 'react';
 import 'rc-slider/assets/index.css';
 
-class LowHealthHealing extends Component {
-  static propTypes = {
-    healEvents: PropTypes.array.isRequired,
-    fightStart: PropTypes.number.isRequired,
-    combatants: PropTypes.object.isRequired,
-  };
+import Combatants from '../../Combatants';
+
+type LowHealthHealingProps = {
+  fightStart: number;
+  combatants: Combatants;
+  healEvents: HealEvent[];
+};
+
+interface ComboHealEvent {
+  /** All abilities in Combo, even duplicates */
+  abilities: Ability[];
+  /** All HealEvents they are based off of */
+  healEvents: HealEvent[];
+  /** totalHeal from all HealEvents */
+  totalEffectiveHeal: number;
+  /** person who was healed */
+  combatant: Combatant;
+  /** HP before the heal */
+  hitPointsBefore: number;
+  /** Max HP for the combo */
+  maxHitPoints: number;
+  /** timestamp of first heal */
+  timestamp: number;
+}
+
+class LowHealthHealing extends React.PureComponent<LowHealthHealingProps> {
   state = {
     maxPlayerHealthPercentage: 0.35,
     minHealOfMaxHealthPercentage: 0.1,
   };
 
+  makeComboEvents(events: HealEvent[], combatants: Combatants) {
+    // first lets group by combatant to make timestamp grouping easier
+    const mappy: Map<Combatant, HealEvent[]> = new Map<Combatant, HealEvent[]>();
+    events.forEach((event) => {
+      const combat = combatants.getEntity(event);
+      if (combat === null) {
+        return;
+      }
+      let array = mappy.get(combat);
+      if (array === undefined) {
+        array = [];
+        mappy.set(combat, array);
+      }
+      array.push(event);
+    });
+
+    const comboMappy: Map<Combatant, ComboHealEvent[]> = new Map<Combatant, ComboHealEvent[]>();
+    // next lets do the timestamp grouping easier
+    mappy.forEach((value, key) => {
+      let array = comboMappy.get(key);
+      if (array === undefined) {
+        array = [];
+        comboMappy.set(key, array);
+      }
+
+      for (let outer = 0; outer < value.length; outer += 1) {
+        const outerEvent = value[outer];
+
+        const currentTimeStamp = outerEvent.timestamp;
+        // small buffer for log weirdness far too low for another cast
+        const upperRange = currentTimeStamp + 100;
```

Comment:
can you make this 100 ms buffer a constant at top of file just so it's clear at a glance what the buffer is